tests/IntervalsTest.php

```diff
@@ -22,6 +22,39 @@ class IntervalsTest extends TestCase
     const INTERVAL_ANY_NODEV = '*';
     const INTERVAL_NONE = '';
 
+    public function testCompactConstraint()
+    {
+        $parser = new VersionParser;
+
+        $original = new MultiConstraint(array(
+            new MultiConstraint(array( // 1.0 - 1.2 || ^1.5
```

Comment:
Your test is keeping exactly one of the original constraint as the compact one. IMO, you should add more tests covering other cases (we also need tests involving dev constraints btw).

Btw, to keep the test more readable, I suggest using the parser to create constraints where possible for most tests.